File: Patrones/Estructurales/Decorator/Python/Azucar.py
from DecoradorBebida import DecoradorBebida

# Clase concreta Azúcar que extiende DecoradorBebida
class Azucar(DecoradorBebida):

    def get_descripcion(self):
        return self._bebida_decorada.get_descripcion() + ", con Azúcar"

    def get_costo(self):
        return self._bebida_decorada.get_costo() + 5.0

File: Patrones/Estructurales/Decorator/Java/src/Azucar.java
public class Azucar extends DecoradorBebida {
    public Azucar(Bebida bebida) {
        super(bebida);
    }

    @Override
    public String getDescripcion() {
        return bebidaDecorada.getDescripcion() + ", con Azúcar";
    }

    @Override
    public double getCosto() {
        return bebidaDecorada.getCosto() + 5.0;
    }
}
